src/css_parser.rs

```diff
@@ -894,6 +928,65 @@ fn parse_color_builtin<'a>(input: &'a str)
     parse_color_no_hash(color)
 }
 
+/// Parse a color of the form 'rgb([0-255], [0-255], [0-255])', without the leading 'rgb(' or
+/// trailing ')'. Alpha defaults to 255.
+fn parse_color_rgb<'a>(input: &'a str)
+-> Result<ColorU, CssColorParseError<'a>>
+{
+    let mut components = input.split(',').map(|c| c.trim());
+    let color = parse_color_rgb_components(&mut components)?;
+    if let Some(arg) = components.next() {
+        return Err(CssColorParseError::ExtraArguments(arg));
+    }
+    Ok(color)
+}
+
+/// Parse a color of the form 'rgba([0-255], [0-255], [0-255], [0.0-1.0])', without the leading
+/// 'rgba(' or trailing ')'.
+fn parse_color_rgba<'a>(input: &'a str)
+-> Result<ColorU, CssColorParseError<'a>>
+{
+    let mut components = input.split(',').map(|c| c.trim());
+    let rgb_color = parse_color_rgb_components(&mut components)?;
+    let a = components.next().ok_or(CssColorParseError::MissingColorComponent("alpha"))?;
+    if a == "" {
+        return Err(CssColorParseError::MissingColorComponent("alpha"));
+    }
+    let a = a.parse::<f32>()?;
+    if a < 0. || a > 1. {
+        return Err(CssColorParseError::FloatValueOutOfRange(a));
+    }
+    let a = (a * 256.).min(255.) as u8;
+    if let Some(arg) = components.next() {
+        return Err(CssColorParseError::ExtraArguments(arg));
+    }
+    Ok(ColorU { a, ..rgb_color })
+}
+
+/// Parse the color components passed as arguments to an rgb(...) CSS color.
+fn parse_color_rgb_components<'a>(components: &mut Iterator<Item = &'a str>)
+-> Result<ColorU, CssColorParseError<'a>>
+{
+    #[inline]
+    fn component_from_str<'a>(components: &mut Iterator<Item = &'a str>, which: &'a str)
+    -> Result<u8, CssColorParseError<'a>>
+    {
+        let c = components.next().ok_or(CssColorParseError::MissingColorComponent(which))?;
+        if c == "" {
+            return Err(CssColorParseError::MissingColorComponent(which));
```

Comment:
I'd prefer if you'd made the `which` in the `MissingColorComponent` an enum instead of a string, i.e `MissingColor::Alpha`, `MissingColor::Red`, etc. - since there can only be four max. This makes the "red", "green", "blue", "alpha" strings part of the display logic.